### PR TITLE
Move the code for taking and saving a screenshot to `RenderingEngine`

### PR DESCRIPTION
**Pull request by red-001**
This PR also fixes the screenshot status messages being untranslatable.

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -1770,25 +1764,17 @@ void Client::makeScreenshot()
...
-				sstr << "Saved screenshot to '" << filename << "'";
-			} else {
-				sstr << "Failed to save screenshot '" << filename << "'";
-			}
-			pushToChatQueue(narrow_to_wide(sstr.str()));
-			infostream << sstr.str() << std::endl;
-			image->drop();
-		}
+		char message_buf[200];
+		snprintf(message_buf, ARRLEN(message_buf), message.c_str(), filename.c_str());
```
Comment (sfan5) [inline src/client.cpp]: you can replace `ARRLEN` with `sizeof` here as char is guaranteed to be 1 byte

File: src/client.cpp [hunk 2]
```diff
@@ -1770,25 +1764,17 @@ void Client::makeScreenshot()
...
-			} else {
-				sstr << "Failed to save screenshot '" << filename << "'";
-			}
-			pushToChatQueue(narrow_to_wide(sstr.str()));
-			infostream << sstr.str() << std::endl;
-			image->drop();
-		}
+		char message_buf[200];
+		snprintf(message_buf, sizeof(message_buf), message.c_str(), filename.c_str());
+		pushToChatQueue(narrow_to_wide(std::string(message_buf)));
```
Comment (nerzhul) [inline src/client.cpp]: why not use wgettext if you want widestring ? it sounds better (don't forget to remove the pointed data)

**Comment by paramat** [comment]
Adoption needed.

File: src/client/renderingengine.cpp [hunk 3]
```diff
@@ -1013,6 +1013,30 @@ const char *RenderingEngine::getVideoDriverFriendlyName(irr::video::E_DRIVER_TYP
...
+bool RenderingEngine::take_screenshot(const std::string &filename)
+{
+	video::IVideoDriver *driver = get_video_driver();
+	irr::video::IImage *const raw_image = driver->createScreenShot();
+
+	if (!raw_image)
+		return false;
+
+	u32 quality = (u32)g_settings->getS32("screenshot_quality");
+	quality = MYMIN(MYMAX(quality, 0), 100) / 100.0 * 255;
```
Comment (SmallJoker) [inline src/client/renderingengine.cpp]: `rangelim(quality, 0, 100)`

At least this variable is now unused in `Client::makeScreenshot`

**Comment by paramat** [comment]
Neglected, closing.

**Pull request closed without merging**